### ISSUE TITLE
[Performance Problem] The tf.image.crop_and_resize op costs much time during training

### ISSUE THREAD

**Issue by Remember2018** [tensorflow/tensorflow#34632]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: None
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 1.15
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.0/7.6.4
- GPU model and memory: GTX 1080Ti, 11Gb

**Problem Description**
Hi Dear experts, my question is about the performance problem on the `tf.image.crop_and_resize` op. My code is as follows:

    def roi_pooling(featuremap, boxes, box_inds, crop_size, data_format='channels_last'):
        if data_format == 'channels_first':
            shp2d = tf.shape(featuremap)[2:]
            featuremap = tf.transpose(featuremap, [0, 2, 3, 1])
        else:
            shp2d = tf.shape(featuremap)[1:3]
        feat_h, feat_w = tf.cast(shp2d[0], tf.float32), tf.cast(shp2d[1], tf.float32)
        xmin, ymin, xmax, ymax = tf.split(boxes, num_or_size_splits=4, axis=1)
        xmin /= feat_w
        ymin /= feat_h
        xmax /= feat_w
        ymax /= feat_h
        normalized_boxes = tf.concat([ymin, xmin, ymax, xmax], axis=1)
        normalized_boxes = tf.stop_gradient(normalized_boxes)
    
        box_features = tf.image.crop_and_resize(image=featuremap,
                                                boxes=normalized_boxes,
                                                box_ind=box_inds,
                                                crop_size=crop_size)  # nhwc
        box_features = tf.layers.average_pooling2d(inputs=box_features, pool_size=2,
                                                   strides=2, padding='valid',
                                                   data_format='channels_last')
    
        if data_format == 'channels_first':
            box_features = tf.transpose(box_features, [0, 3, 1, 2])  # nhwc --> nchw
    
        return box_features
    
And I've take the full trace of the training procedure using the `tensorflow.python.client.timeline` object and got the following tracing graph. 
![A0587B57-0D3A-49ba-81C9-56DC9332CB57](https://user-images.githubusercontent.com/43327429/69691790-01c19e00-110b-11ea-8eb3-85e3988eb209.png)

From this graph, we can see that the `tf.image.crop_and_resize` ops costs much time during one training step. So is this reasonable? Could you please give some advice on how to improve the performance? Thanks.

**Comment by oanush** [comment]
@Remember2018 ,
Can you provide a standalone code to reproduce the issue reported here as the given code is not sufficient to replicate it?Thanks!

**Comment by Remember2018** [comment]
@oanush thanks a lot for your kind reply! I've found the cause. In the project, the number of input `boxes` of `tf.image.crop_and_resize` are about 12000, which seems too large and leads to slow performance. When decreasing the number, we can get the reasonable performance.
